### PR TITLE
Dg/talent_course_dates_2-6670

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

Add date ranges where TalentLMS courses will be considered active. Only active courses will be included in the required training for individual users.

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [X] I have provided testing instructions in this PR or the related issue (or not applicable)

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected

### CODE REVIEW

**Comment by dtgreiner** [comment]
Screenshot 2024-12-02 at 9 09 24 AM

**Comment by dtgreiner** [comment]
This is a recreation of [this PR](https://github.com/greenriver/hmis-warehouse/pull/4969).

File: app/models/talentlms/course.rb [hunk 1]
```diff
@@ -68,5 +85,31 @@ def completed_by?(user)
...
+
+    def active_date_order_value
+      [
+        active? ? 0 : 1,
+        start_date ? 0 : 1,
+        start_date,
+        end_date ? 0 : 1,
+        end_date,
+      ]
+    end
```
Comment (eanders) [inline app/models/talentlms/course.rb:97]: This seems overly complex.  What about a scope, something like:
``
scope :ordered, -> do
 order(active: :desc, Arel.sql(arel_table[:start_date].desc.to_sql + ' NULLS LAST'), Arel.sql(arel_table[:end_date].desc.to_sql + ' NULLS LAST'))
end
```

That's probably non-functional, and there's probably a better way to indicate nulls last, but then the call laster can just be `config.courses.ordered`.

That said, if it's not easy, and this seems to work, we need to instantiate all of the courses anyway, so it's not a big deal.

Comment (eanders) [reply app/models/talentlms/course.rb:97]: Oh, and I see, `active?` isn't a column, but a method....

**Review by eanders — APPROVED** [review]
Thanks @dtgreiner !

**Pull request merged**